backend/danswer/connectors/teams/connector.py

```diff
@@ -0,0 +1,288 @@
+        first_poster = (
+            top_message.properties.get("from", {})
+            .get("user", {})
+            .get("displayName", {})
+        )
+        channel_name = channel.properties["displayName"]
+        thread_subject = top_message.properties["subject"]
+        snippet = parse_html_page_basic(
+            top_message.body.content[:50].rstrip() + "..."
+            if len(top_message.body.content) > 50
+            else top_message.body.content
+        )
+        if not first_poster or not channel_name or not thread_subject or not snippet:
+            return None
+
+        return f"{first_poster} in {channel_name} about {thread_subject}: {snippet}"
+
+    def _convert_thread_to_document(
+        self,
+        channel: Channel,
+        thread: list[ChatMessage],
+    ) -> Document | None:
+        if len(thread) <= 0:
+            return None
+
+        most_recent_message_datetime: datetime | None = None
+        top_message = thread[0]
+        post_members_list: list[BasicExpertInfo] = []
+        thread_text = ""
+
+        sorted_thread = sorted(thread, key=get_created_datetime, reverse=True)
+
+        if sorted_thread:
+            most_recent_message = sorted_thread[0]
+            most_recent_message_datetime = datetime.strptime(
+                most_recent_message.properties["createdDateTime"],
+                datetime_format_string,
+            )
+
+        for message in thread:
+            # add text and a newline
+            if message.body.content:
+                message_text = parse_html_page_basic(message.body.content)
+                thread_text += message_text
+
+            # if it has a subject, that means its the top level post message, so grab its id, url, and subject
+            if message.properties["subject"]:
+                top_message = message
+
+            # check to make sure there is a valid display name
+            if message.properties["from"]:
+                if message.properties["from"]["user"]:
+                    if message.properties["from"]["user"]["displayName"]:
+                        message_sender = message.properties["from"]["user"][
+                            "displayName"
+                        ]
+                        # if its not a duplicate, add it to the list
+                        if message_sender not in [
+                            member.display_name for member in post_members_list
+                        ]:
+                            post_members_list.append(
+                                BasicExpertInfo(display_name=message_sender)
+                            )
+
+        # if there are no found post members, grab the members from the parent channel
+        if not post_members_list:
+            post_members_list = self._extract_channel_members(channel)
+
+        if not thread_text:
+            return None
+
+        semantic_string = self._construct_semantic_identifier(channel, top_message)
+        if not semantic_string:
```

Comment:
Slight preference to not drop messages due to this, you can probably do something like "Unknown user in Unknown channel" something like that